drivers/windows/overlay/peerdb_windows.go

```diff
@@ -66,9 +75,15 @@ func (d *driver) peerAdd(nid, eid string, peerIP net.IP, peerIPMask net.IPMask,
 			return err
 		}
 
+		if windowsBuild == 14393 {
+			peerMu.Lock()
+		}
```

Comment:
`removeEndpointWithAddress` does a bunch of stuff and also has a lock;  I am a little bit concerned wrapping the whole call into a lock.
Why not just use the mutex and `windowsBuild` and use them in `ov_endpoint_windows.go`, we are still in the same package and the global variable should still work.
I also suggest to expand the lock to all these calls in `ov_endpoint_windows.go`
@mavenugo WDYT

```
ov_endpoint_windows.go:         _, err := hcsshim.HNSEndpointRequest("DELETE", networkEndpoint.profileID, "")
ov_endpoint_windows.go:         _, err := hcsshim.HNSEndpointRequest("DELETE", ep.profileID, "")
ov_endpoint_windows.go: hnsresponse, err := hcsshim.HNSEndpointRequest("POST", "", string(configurationb))
ov_endpoint_windows.go:         hcsshim.HNSEndpointRequest("DELETE", hnsresponse.Id, "")
ov_endpoint_windows.go: _, err := hcsshim.HNSEndpointRequest("DELETE", ep.profileID, "")
```

Reply:
@pradipd I agree with @dani-docker it is safer to scope the lock around `HNSEndpointRequest`.  In this specific case, there is also another `n.Lock` inside the said function and it can easily cause deadlock if we are not careful. Also, I do understand that this issue will happen ONLY on remote Peer add/delete case. Hence I will leave it to your best judgement on where to scope this lock to (To all the instances as @dani-docker pointed out, or just the remote peer add/delete cases).

Reply:
I think we will be OK in removeEndpointWithAddress.  The network lock is only held when modifying the endpoint table.  There are no other functions being called.

However, I do think your suggestion is a good idea.  However, I'd prefer to do that in separate PR.  We know we have an issue with remote endpoints and we have a repro.  I'd prefer to fix this ASAP, and look into moving it separately.  I also feel adding it to more places is just going to increase our chances of a deadlock.